python/src/main.cpp

```diff
@@ -24,31 +26,35 @@ opAlgoDispatchPyInit(std::shared_ptr<kp::Algorithm>& algorithm,
                  push_consts.size(),
                  std::string(py::str(push_consts.dtype())));
 
-    if (push_consts.dtype().is(py::dtype::of<std::float_t>())) {
-        std::vector<float> dataVec((float*)info.ptr,
-                                   ((float*)info.ptr) + info.size);
-        return std::unique_ptr<kp::OpAlgoDispatch>{ new kp::OpAlgoDispatch(
-          algorithm, dataVec) };
-    } else if (push_consts.dtype().is(py::dtype::of<std::uint32_t>())) {
-        std::vector<uint32_t> dataVec((uint32_t*)info.ptr,
-                                      ((uint32_t*)info.ptr) + info.size);
-        return std::unique_ptr<kp::OpAlgoDispatch>{ new kp::OpAlgoDispatch(
-          algorithm, dataVec) };
-    } else if (push_consts.dtype().is(py::dtype::of<std::int32_t>())) {
-        std::vector<int32_t> dataVec((int32_t*)info.ptr,
-                                     ((int32_t*)info.ptr) + info.size);
-        return std::unique_ptr<kp::OpAlgoDispatch>{ new kp::OpAlgoDispatch(
-          algorithm, dataVec) };
-    } else if (push_consts.dtype().is(py::dtype::of<std::double_t>())) {
-        std::vector<double> dataVec((double*)info.ptr,
-                                    ((double*)info.ptr) + info.size);
-        return std::unique_ptr<kp::OpAlgoDispatch>{ new kp::OpAlgoDispatch(
-          algorithm, dataVec) };
-    } else {
-        throw std::runtime_error("Kompute Python no valid dtype supported");
-    }
+    std::vector<byte> dataVec((byte*)info.ptr,
+                                 ((byte*)info.ptr) + info.size);
+    return std::unique_ptr<kp::OpAlgoDispatch>{ new kp::OpAlgoDispatch(
+      algorithm, dataVec) };
 }
 
+class PyTypeContainer : public ABCTypeContainer
```

Comment:
Reading this it seems that this new iteration of the implementation was added to explore the implementation on the python side, but it seems this is also adding quite a lot of complexity on the C++ side (also there's quite a few python-internal references here that I'm not sure about) - I would be keen to explore how complexity can be reduced in this implementation